### ISSUE TITLE
Installation issue: gcc@9.5.0

### ISSUE THREAD

**Issue by baloe** [LLNL/spack#32286]
### Steps to reproduce the issue

```console
$ spack install gcc@9.5.0
```

### Error message

<details><summary>Error message</summary><pre>
[...]
libtool: link: ranlib .libs/libitm.a
libtool: link: ( cd ".libs" && rm -f "libitm.la" && ln -s "../libitm.la" "libitm.la" )
make[4]: Leaving directory '/tmp/bastian/spack-stage/spack-stage-gcc-9.5.0-3a6ryytdjitmnqokyl635p4uadxuv5e4/spack-src/spack-build/x86_64-pc-linux-gnu/libitm'
make[3]: Leaving directory '/tmp/bastian/spack-stage/spack-stage-gcc-9.5.0-3a6ryytdjitmnqokyl635p4uadxuv5e4/spack-src/spack-build/x86_64-pc-linux-gnu/libitm'
make[2]: Leaving directory '/tmp/bastian/spack-stage/spack-stage-gcc-9.5.0-3a6ryytdjitmnqokyl635p4uadxuv5e4/spack-src/spack-build/x86_64-pc-linux-gnu/libitm'
make[1]: Leaving directory '/tmp/bastian/spack-stage/spack-stage-gcc-9.5.0-3a6ryytdjitmnqokyl635p4uadxuv5e4/spack-src/spack-build'
make: *** [Makefile:996: all] Error 2
</pre></details>

### Information on your system

* **Spack:** 0.19.0.dev0 (dab46296b6591ca35dd8986bc474b3c7eb08edcd)
* **Python:** 3.10.5
* **Platform:** linux-manjaro21-skylake
* **Concretizer:** clingo

### Additional information

These dependencies were installed successfully, before failing to build gcc:
```
$ spack find
==> 21 installed packages
-- linux-manjaro21-skylake / gcc@12.1.1 -------------------------
autoconf@2.69    berkeley-db@18.1.40  diffutils@3.8  gdbm@1.19  libelf@0.8.13  libsigsegv@2.13  m4@1.4.19  mpfr@3.1.6   perl@5.34.1    readline@8.1.2  zlib@1.2.12
automake@1.16.5  bzip2@1.0.8          gawk@5.1.1     gmp@6.2.1  libiconv@1.16  libtool@2.4.7    mpc@1.1.0  ncurses@6.3  pkgconf@1.8.0  texinfo@6.5
```

[spack-build-env.txt](https://github.com/spack/spack/files/9386378/spack-stage-gcc-9.5.0-3a6ryytdjitmnqokyl635p4uadxuv5e4_spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9386379/spack-stage-gcc-9.5.0-3a6ryytdjitmnqokyl635p4uadxuv5e4_spack-build-out.txt)

@alalazo  @michaelkuhn

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by michaelkuhn** [comment]
This seems to be caused by newer glibc versions (see 2.1 at https://sourceware.org/glibc/wiki/Release/2.36). Do newer versions of GCC work?

**Comment by baloe** [comment]
> This seems to be caused by newer glibc versions (see 2.1 at https://sourceware.org/glibc/wiki/Release/2.36). Do newer versions of GCC work?

I don't know about newer versions, I'm travelling right now, that makes it a bit difficult to let my laptop compile stuff for 2 hours just to see if an error occurs after 2 hours.
But I really need GCC<10.

**Comment by baloe** [comment]
> This seems to be caused by newer glibc versions (see 2.1 at https://sourceware.org/glibc/wiki/Release/2.36). Do newer versions of GCC work?

Can't spack install an older  version of glibc and use that one?

**Comment by haampie** [comment]
edit: yes, but it's not a path worth pursuing: https://github.com/spack/spack/wiki/Building-gcc-with-glibc-together-in-10-easy-steps

**Comment by haampie** [comment]
Closed by #35798
